sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1792,4 +1804,82 @@ private synchronized void register(Class<? extends PipelineOptions> iface) {
       return combinedCache.get(interfaces).getPropertyDescriptors();
     }
   }
+
+  /**
+   * @param prefix a prefix filter on the map keys.
+   * @param options the options to convert to a pipeline options.
+   * @return a pipeline options instance based on the specified configuration.
+   */
+  @Experimental
+  public static PipelineOptions fromMap(final String prefix, final Map<String, String> options) {
```

Comment:
Why is this important to have part of the API?

It seems like we should ask users to filter their map keys themselves.